javascript/src/consent.ts

```diff
@@ -2,6 +2,38 @@
 This module implements Humbug's user consent mechanisms.
 */
 
-type Yes = "1" | "t" | "y" | "T" | "Y" | "true" | "yes" | "True" | "Yes" | "TRUE" | "YES"
-type No = "0" | "f" | "n" | "F" | "N" | "false" | "no" | "False" | "No" | "FALSE" | "NO"
+export default class HumbugConsent {
+    buggerOffStatus: boolean
 
+    constructor(
+        public mechanisms?: boolean
+    ) {
+        if (mechanisms === undefined) {
+            this.mechanisms = false
+        }
+        this.buggerOffStatus = environmentVariableOptOut()
+    }
+
+    check(): boolean {
+        if (this.mechanisms === false) {
+            return false
+        }
+        else if (this.buggerOffStatus === false) {
+            return false
+        }
+        return true
+    }
+}
+
+function environmentVariableOptOut(): boolean {
```

Comment:
`BUGGER_OFF=true` is not for libraries integrating with humbug, it is for the end user.

If someone who is using `humbug` in their library wants to have `environmentVariableOptOut`, we should implement similiarly as in `humbug/python`.

Reply:
This should instead be like this:
```javascript
function environmentVariableOptOut(envVar: string, optOutValues: Array<string>): func():bool {
    return func(): bool {
        const envVal = process.env[envVar];
        return optOutValues.includes(envVal);
    }
}
```